frontend/kesaseteli/employer/src/components/application/form/DateInput.tsx

```diff
@@ -0,0 +1,116 @@
+import useApplicationFormField from 'kesaseteli/employer/hooks/application/useApplicationFormField';
+import { useTranslation } from 'next-i18next';
+import React from 'react';
+import {
+  RegisterOptions,
+  useFormContext,
+  UseFormRegister,
+} from 'react-hook-form';
+import {
+  $GridCell,
+  GridCellProps,
+} from 'shared/components/forms/section/FormSection.sc';
+import useLocale from 'shared/hooks/useLocale';
+import Application from 'shared/types/employer-application';
+import {
+  convertToBackendDateFormat,
+  convertToUIDateFormat,
+  isValidDate,
+  parseDate,
+} from 'shared/utils/date.utils';
+import { isEmpty } from 'shared/utils/string.utils';
+
+import { $DateInput } from './DateInput.sc';
+
+type Props = {
+  validation: RegisterOptions<Application>;
+  id: NonNullable<Parameters<UseFormRegister<Application>>[0]>;
+} & GridCellProps;
+
+// TODO: This can be removed after backend supports invalid values in draft save
+const convertDateForBackend = (dateString: string): string | undefined => {
+  const result = convertToBackendDateFormat(dateString);
+  return isEmpty(result) ? undefined : result;
+};
+
+const DateInput = ({
+  id,
+  validation,
+  ...$gridCellProps
+}: Props): ReturnType<typeof HdsDateInput> => {
+  const { t } = useTranslation();
+  const locale = useLocale();
+  const { register } = useFormContext<Application>();
+  const {
+    fieldName,
+    getValue,
+    getError,
+    setError,
+    clearErrors,
+    setValue,
+    clearValue,
+  } = useApplicationFormField<string>(id);
+
+  const date = convertToUIDateFormat(getValue());
+
+  const validate = React.useCallback(
+    (value) => isValidDate(parseDate(value)),
+    []
+  );
+
+  const errorType = getError()?.type;
+  const errorText = React.useMemo((): string | undefined => {
+    if (!errorType) {
+      return undefined;
+    }
+    if (['pattern', 'required'].includes(errorType)) {
+      return `${t(`common:application.form.errors.${errorType}`)}. ${t(
+        `common:application.form.helpers.date`
+      )}`;
+    }
+    return t(`common:application.form.errors.${errorType}`);
+  }, [t, errorType]);
+
+  // TODO: This can be removed after backend supports invalid values in draft save
+  const handleChange = React.useCallback(
+    (dateString: string) => {
+      const uiDate = convertToUIDateFormat(dateString);
+      if (isEmpty(uiDate)) {
+        setError('pattern');
+        clearValue();
+      } else {
+        clearErrors();
+        setValue(uiDate);
+      }
+    },
+    [setError, clearValue, setValue, clearErrors]
+  );
+
+  return (
+    <$GridCell {...$gridCellProps}>
+      <$DateInput
+        {...register(id, {
+          ...validation,
+          validate,
+          setValueAs: convertDateForBackend,
+        })}
+        key={id}
+        id={id}
+        data-testid={id}
+        name={id}
+        required={Boolean(validation.required)}
+        initialMonth={new Date()}
+        defaultValue={date}
```

Comment:
I noticed that this component has the `defaultValue` prop but not the `value`. Is that intentional?

Reply:
I did lot of things back and forth. I think it's an accident. I replaced this with `value={date}` and it seem to work in the same way